packages/ui/v2/core/form/select/Select.tsx

```diff
@@ -1,106 +1,83 @@
 import { useId } from "@radix-ui/react-id";
 import * as React from "react";
 import ReactSelect, {
-  components,
+  components as reactSelectComponents,
   GroupBase,
-  InputProps,
   Props,
   SingleValue,
   MultiValue,
-  OptionProps,
+  SelectComponentsConfig,
 } from "react-select";
 
 import classNames from "@calcom/lib/classNames";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
-import { Icon } from "@calcom/ui/Icon";
 
-import { Label } from "./fields";
+import { Label } from "../fields";
+import {
+  ControlComponent,
+  InputComponent,
+  MenuComponent,
+  MenuListComponent,
+  OptionComponent,
+  SingleValueComponent,
+  ValueContainerComponent,
+} from "./components";
 
 export type SelectProps<
   Option,
   IsMulti extends boolean = false,
   Group extends GroupBase<Option> = GroupBase<Option>
 > = Props<Option, IsMulti, Group>;
 
-export const InputComponent = <Option, IsMulti extends boolean, Group extends GroupBase<Option>>({
-  inputClassName,
-  ...props
-}: InputProps<Option, IsMulti, Group>) => {
-  return (
-    <components.Input
-      // disables our default form focus hightlight on the react-select input element
-      inputClassName={classNames("focus:ring-0 focus:ring-offset-0", inputClassName)}
-      {...props}
-    />
-  );
-};
-
-const OptionComponent = <Option, IsMulti extends boolean, Group extends GroupBase<Option>>({
+export const getReactSelectProps = <
```

Comment:
I'm not happy with all the generics defined in this file, but otherwise `react-select` wouldn't stop complaining. Tried to abstract it into a function, but that would make it a lot more complicated for a few lines less code.. If anyone has a great idea — always welcome!